### PR TITLE
Session 05 HW

### PR DESCRIPTION
**Pull request by robalford**

### CODE REVIEW

File: students/robalford/session05/.cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (PythonCHB) [inline students/robalford/session05/.cache/v/cache/lastfailed]: this shouldn't have been added to git.

**Comment by PythonCHB** [comment]
looks like a bunch of cache and temp files got added to git -- pytest does add those.

beware "git add *" ! -- you want to be careful about what you add. I usually do thinks like:

git add *.py *.txt

if I have a bunch to add.

you'll need to remove them with:

git rm __pycache__/*

etc....

File: students/robalford/session05/comprehensions.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# 1
+
+food_prefs = {"name": u"Chris",
+              u"city": u"Seattle",
+              u"cake": u"chocolate",
+              u"fruit": u"mango",
+              u"salad": u"greek",
+              u"pasta": u"lasagna"}
+
+# Is this the preferred (Pythonic) style for wrapping long lines of text?
```
Comment (PythonCHB) [inline students/robalford/session05/comprehensions.py]: yes, this is common style. You can also use a triple quoted string:

person_likes_food = """{name} is from {city}, and he likes {cake} cake,
{fruit} fruit, {salad} salad and {pasta} pasta.""".format(**food_prefs)

though that will put in a newline.

File: students/robalford/session05/comprehensions.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# s2 = {number for number in range(21) if number % 2 == 0}
+# s3 = {number for number in range(21) if number % 3 == 0}
+# s4 = {number for number in range(21) if number % 4 == 0}
+
+# sets = []
+
+# for num in range(2, 5):
+#     sets.append({number for number in range(21) if number % num == 0})
+
+sets = [{number for number in range(21) if number % num == 0} for num in range(2, 5)]
```
Comment (PythonCHB) [inline students/robalford/session05/comprehensions.py]: nice!

File: students/robalford/session05/test_coding_bat.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
+from coding_bat import string_splosion, cigar_party, count_evens
```
Comment (PythonCHB) [inline students/robalford/session05/test_coding_bat.py]: you got the idea -- nice tests.

**Comment by robalford** [comment]
Should I add .pyc to my .gitignore as well to make it explicit that these
kinds of files should never be tracked by git? Or are there any
circumstances where you would want to add them?

**Comment by PythonCHB** [comment]
yes, adding *.pyc to .gitignore is a good idea.

**Comment by cewing** [comment]
Just an FYI, github maintains a pretty sensible [Python .gitignore file](https://github.com/github/gitignore/blob/master/Python.gitignore).  When you start a new repository, it's an option to begin with one in place, and even for existing repos, adding it by pasting what you find at that link is quite a good starting place.

**Comment by PythonCHB** [comment]
Thanks Cris,

I hadn't noticed that.

And I'll also point you to this repo:

https://github.com/github/gitignore

Which has comprehensive .gitignore files for all kinds of projects.

Chris

On Tuesday, November 10, 2015, Cris Ewing <notifications@github.com> wrote:

> Just an FYI, github maintains a pretty sensible Python .gitignore file
> <https://github.com/github/gitignore/blob/master/Python.gitignore>. When
> you start a new repository, it's an option to begin with one in place, and
> even for existing repos, adding it by pasting what you find at that link is
> quite a good starting place.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UWPCE-PythonCert/IntroPython2015/pull/83#issuecomment-155576242>
> .
>


-- 
Christopher Barker, PhD

Python Language Consulting
  - Teaching
  - Scientific Software Development
  - Desktop GUI and Web Development
  - wxPython, numpy, scipy, Cython

**Pull request merged**